bugsbunnyctf2017-rev150/solve.py

```diff
@@ -0,0 +1,61 @@
+from manticore.native import Manticore
+import sys
+
+prog = sys.argv[1]
+params = sys.argv[2:]
+
+"""
+To run:
+$ python rev150.py ./rev150 00000000000000000000
+
+The symbolic execution will overwrite the integer value with some number from 0 to 99999999999999999999 (you know that there are only 20 digits from IDA Pro. The program increments until the correct value is found.
+
+For testing purposes, the solution is 42813724579039578812.
+
+Run time: 9:10.62
+"""
+
+m = Manticore(prog, params)
+
+"""
+Here, we are setting the initial password (from IDA Pro, we can deduce a majority of the digits). We know that the password consists of 20 digits, all of which must be numeric.
+"""
+with m.locked_context() as context: 
+    context['password'] = 42810720579039578812
+
+@m.hook(0x401be2)
+def inject_password(state):
+    """
+    At this point, we inject our chosen password into the address holding the password inputted to the
+    binary. The password is formatted to be 20 digits long.
+    """
+    with m.locked_context() as context:
+        print("[+] injecting password: " + str(format(context['password'], '020')))
+        state.cpu.write_bytes(state.cpu.RDI,str(format(context['password'],'020')))
+        
+@m.hook(0x401e5a)
+def failed(state):
+    """
+    If the password is incorrect, we will increment the password (so long as it remains 20 digits) and
+    return to the original point of injection.
+    """
+    with m.locked_context() as context:
+        if (len(str(context['password'])) == 20):
+            context['password'] += 1000000000000
+            state.cpu.RIP = 0x401be2
+            print("[-] incorrect password")
+        else:
+            print("[-] no password found")
+            m.terminate()
+
+@m.hook(0x401e49) 
+def success(state):
+    """
+    If our password passes all of the checks, we can return it as the flag.
+    """
+    with m.locked_context() as context:
+        print("[+] success. flag: BugsBunny{" + str(context['password']) + "}")
```

Comment:
It is more readable if we use proper string formatting instead of concatenating strings. In Python, there are three ways for that currently:
* `%` operator for string, e.g.: `'some int: %d' % some_int` - note, that if you want to format more than one thing you need to pass a collection e.g. `'some int: %d and string: %s' % (some_int, some_str)`
* `str.format` method e.g.: `'some int: {} some str: {}'.format(some_int, some_str)`
* f-strings: `f'some int: {some_int} some str: {some_str}'`

The % operator and `.format` methods are old ways to format strings. You can learn more about them on  https://pyformat.info/. The f-string is a [Literal String Interpolation added to Python 3.6 proposed in PEP-498](https://www.python.org/dev/peps/pep-0498/).

I'd go for changing it to f-strings since we use them in Manticore anyway (and so we require Python>=3.6):
```suggestion
        print(f"[+] success. flag: BugsBunny{{context['password']}}")
```